tests/unit_tests/test_refactor.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+def test_newest():
+    item_a = Clothing(age=1.0)
+    item_b = Decor(age=2.0)
+    item_c = Clothing(age=4.0)
+    item_d = Decor(age=5.0)
+    item_e = Clothing(age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    best_item = tai.get_newest_item()
+
+    assert best_item.category == "Clothing"
+    assert best_item.age == pytest.approx(1.0)
```

Comment:
Love that you've added these extra tests!

One small thing, it looks like many of your tests have the newest item first. Code that just selects the first item in the list could pass them without actually implementing the correct logic. Consider placing the newer item somewhere in the middle of the list more frequently to avoid these false passes. I see you're doing this in `test_newest_reordered`, which is great!